client/request/request.go

```diff
@@ -152,11 +154,23 @@ func Req(opts *Options) (*http.Response, error) {
 	if err != nil {
 		// for a development release, we fallback on http mode if the https request
 		// is not successful.
-		// TODO: better identify the error (cross-platform)
 		if config.IsDevRelease() && scheme == "https" {
-			logrus.Debug("[request] fallback on http transport since https request has failed", err)
-			atomic.StoreInt32(&httpFallback, 1)
-			return Req(opts)
+			var isConnRefused bool
+			switch t := err.(type) {
+			case *net.OpError:
+				if t.Op == "read" {
+					isConnRefused = true
+				}
+			case syscall.Errno:
+				if t == syscall.ECONNREFUSED {
+					isConnRefused = true
+				}
+			}
```

Comment:
It looks complicated and I'm not sure that we are not missing some cases. For example, if we connect in https to the admin port (http), I think we will another error to say that tle TLS handshake has failed.